pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -200,6 +210,32 @@ type PipelineRunStatus struct {
 	PipelineRunStatusFields `json:",inline"`
 }
 
+// PipelineRunSucceededReason represents a reason for the pipeline run "Succeeded" condition
+type PipelineRunSucceededReason string
+
+const (
+	// PipelineRunSucceededReasonStarted is the reason set when the PipelineRun has just started
+	PipelineRunSucceededReasonStarted PipelineRunSucceededReason = "Started"
+	// PipelineRunSucceededReasonRunning is the reason set when the PipelineRun is running
+	PipelineRunSucceededReasonRunning PipelineRunSucceededReason = "Running"
+	// PipelineRunSucceededReasonSuccessful is the reason set when the PipelineRun completed successfully
+	PipelineRunSucceededReasonSuccessful PipelineRunSucceededReason = "Succeeded"
+	// PipelineRunSucceededReasonCompleted is the reason set when the PipelineRun completed successfully with one or more skipped Tasks
+	PipelineRunSucceededReasonCompleted PipelineRunSucceededReason = "Completed"
+	// PipelineRunSucceededReasonFailed is the reason set when the PipelineRun completed with a failure
+	PipelineRunSucceededReasonFailed PipelineRunSucceededReason = "Failed"
+	// PipelineRunSucceededReasonCancelled is the reason set when the PipelineRun cancelled by the user
+	// This reason may be found with a corev1.ConditionFalse status, if the cancellation was processed successfully
+	// This reason may be found with a corev1.ConditionUnknown status, if the cancellation is being processed or failed
+	PipelineRunSucceededReasonCancelled PipelineRunSucceededReason = "Cancelled"
+	// PipelineRunSucceededReasonTimedOut is the reason set when the PipelineRun has timed out
+	PipelineRunSucceededReasonTimedOut PipelineRunSucceededReason = "PipelineRunTimeout"
+)
```

Comment:
I love the way its all in one place now and inside of `types` instead of part of `resources` 🙏